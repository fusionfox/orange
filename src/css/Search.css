.Search__Title {
  text-align: center;
  margin-top: 50px;
}
