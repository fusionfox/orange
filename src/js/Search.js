import React, { Component } from 'react'
import axios from 'axios'
import './../css/Search.css'
import Title from './components/Title'
import SearchControls from './components/SearchControls'
import ImageGrid from './components/ImageGrid'

axios.defaults.baseURL = 'https://images-api.nasa.gov'

class Search extends Component {
  constructor() {
    super()
    this.state = {
      images: []
    }
  }

  nasaDataAdapter = (nasaItem) => {
    return ({
      id: nasaItem.data[0].nasa_id,
      alt: nasaItem.data[0].title,
      href: nasaItem.links[0].href
    })
  }

  searchNasa = (query) => {
    axios.get('/search?q=moon')
      .then((response) => response.data.collection.items.map(this.nasaDataAdapter))
      .then((data) => this.setState({images: data}))
      .catch((error) => {
        console.log(error);
      });
  }

  render() {
    return (
      <div className="Search">
        <Title className="Search__Title" text='Search Page' />
        <SearchControls className="Search__Controls" performSearch={this.searchNasa}/>
        <ImageGrid className="Search__ImageGrid" images={this.state.images} />
      </div>
    );
  }
}

export default Search;
