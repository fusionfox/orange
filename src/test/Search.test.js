import React from 'react'
import renderer from 'react-test-renderer'
import Search from '../js/Search'

test('Search rendered correctly', () => {
  const component = renderer.create(
    <Search />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});
