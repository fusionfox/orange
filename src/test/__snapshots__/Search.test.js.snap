// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search rendered correctly 1`] = `
<div
  className="Search"
>
  <h1
    className="Search__Title Title"
  >
    Search Page
  </h1>
  <div
    className="Search__Controls SearchControls"
  >
    <div
      className="SearchControls__query"
    >
      <input
        onKeyUp={[Function]}
        placeholder="Input query then press Enter"
        type="text"
      />
    </div>
    <div
      className="SearchControls__filters"
    >
      <label>
        <input
          checked={true}
          name="media_type"
          onChange={[Function]}
          type="radio"
          value="image"
        />
        Images
      </label>
      <label>
        <input
          checked={false}
          name="media_type"
          onChange={[Function]}
          type="radio"
          value="audio"
        />
        Audio
      </label>
    </div>
  </div>
  <ul
    className="Search__ImageGrid ImageGrid"
  />
</div>
`;
